_posts/2024-07-01-zio-fullstack-webapp.md

```diff
@@ -0,0 +1,844 @@
+        Form.renderVar(personVar)                                         // (3)
+      ),
+      div(
+        styleAttr := "width: 100px; float: left; margin-top: 200px;",
+        Button(
+          "-- Post -->",
+          onClick --> { _ =>
+            // scalafmt:off
+
+            PersonEndpoint                                                // (4)
+              .createEndpoint(personVar.now())
+              .emitTo(userBus) 
+
+            // scalafmt:on
+
+          }
+        )
+      ),
+      div(
+        styleAttr := "width: 600px; float: left;",
+        h1("Databinding"),
+        child.text <-- personVar.signal.map(p => s"$p"),                  // (5)
+        h1("Response"),
+        child <-- userBus.events.map(renderUser)                          // (6)
+      )
+    )
+
+  def renderUser(user: User) =
+    div(
+      h2("User"),
+      div(s"Name: ${user.name}"),
+      div(s"Age: ${user.age}"),
+      div(s"Pet: ${user.pet}"),
+      div(s"Creation Date: ${user.creationDate}")
+    )
+```
+
+* **(1)** A Laminar Var that will hold the Person to create
+* **(2)** An EventBus that will receive the User created
+* **(3)** A Form that will bind the Person to the UI see [here](https://cheleb.github.io/laminar-form-derivation/docs/index.html)
+* **(4)** A Button that will call the PersonEndpoint.createEndpoint with the current value of the Person Var
+* **(5)** A div that will display the current value of the Person Var (debug purpose)
+* **(6)** A div that will display the User created, reacting to the EventBus events.
+
+This is a lot of magic in a single page.
+
+
+# Interacting with JavaScript ecosystem
+
+We can leverage the JavaScript ecosystem in a type safe manner. 
+
+Either by using Scala.js facade or by using the JavaScript API directly.
+
+To illustrate this,  we are using a Laminar binding to a SAP UI5 component.
+
+But more interestingly, you will find in the project a facade to the [Chart.js](https://www.chartjs.org/) library.
+
+This being is generated by [Scalablytyped](https://scalablytyped.org/), a tool that will generate Scala.js facade from TypeScript definition.
+
+This sample is a shamy cut and paste from Sébastien Doeraene demonstration: [Getting started with Scala.js, Laminar and ScalablyTyped](https://www.youtube.com/watch?v=UePrOa_1Am8)
+
+```sbt
+addSbtPlugin("org.scalablytyped.converter" % "sbt-converter" % "1.0.0-beta44")
+```
+
+Some configuration, devDependencies point to TypeScript whereas runtime dependencies point javascript:
+```json
+{
+  "name": "scalablytyped",
+  "private": true,
+  "version": "0.0.1",
+  "type": "module",
+  "dependencies": {
+    "chart.js": "2.9.4"
+  },
+  "devDependencies": {
+    "typescript": "5.4.5",
+    "@types/chart.js": "2.9.29"
+  }
+}
+```
+
+In this example Chart.js binding are generated from the TypeScript definition and fully accessible from ScalaJS at compile time, whereas the chart.js will be used at runtime.
```

Comment:
bindings are...

In general, I saw the need for a lot of plural 's' here in the article